packages/tools/nodeGeometryEditor/src/components/preview/previewManager.ts

```diff
@@ -349,9 +351,9 @@ export class PreviewManager {
                 this._updateStandardMaterial();
                 this._setMaterial();
                 this._mesh.useVertexColors = true;
-                this._mesh.onAfterRenderObservable.addOnce(() => {
-                    this._toDelete?.dispose();
-                    this._toDelete = null;
+                this._mesh.onMeshReadyObservable.addOnce(() => {
+                    this._toDelete.forEach((m) => m.dispose());
+                    this._toDelete = [];

```

Comment:
nit - `.length = 0`?
```suggestion
                    this._toDelete.length = 0;
```